homeassistant/components/intellifire/__init__.py

```diff
@@ -17,17 +20,29 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up IntelliFire from a config entry."""
     LOGGER.debug("Setting up config entry: %s", entry.unique_id)
 
-    # Define the API Object
-    api_object = IntellifireAsync(entry.data[CONF_HOST])
+    # Define the API Objects
+    read_object = IntellifireAsync(entry.data[CONF_HOST])
+
+    ift_control = IntellifireControlAsync(
+        fireplace_ip=entry.data[CONF_HOST],
+    )
+    try:
+        await ift_control.login(
+            username=entry.data[CONF_USERNAME],
+            password=entry.data[CONF_PASSWORD],
+        )
+    except (ConnectionError, ClientConnectionError, LoginException) as err:
+        raise ConfigEntryAuthFailed(err) from err
```

Comment:
ConnectionError and ClientConnectionError should raise a `ConfigEntryNotReady` so it is retried later. Only `LoginException` should trigger re-auth flow.